ldms/src/sampler/spank/slurm_notifier.c

```diff
@@ -663,35 +775,45 @@ jbuf_t make_task_exit_data(spank_t sh, const char *event, const char *context)
 	return jb;
 }
 
-static int nnodes(spank_t sh)
-{
-	uint32_t nnodes;
-	spank_err_t err = spank_get_item(sh, S_JOB_NNODES, &nnodes);
-	if (err)
-		return 0;
-	return nnodes;
-}
-
 int slurm_spank_init(spank_t sh, int argc, char *argv[])
 {
 	spank_context_t context = spank_context();
-	const char *context_str;
-	jbuf_t jb;
-
-	if (0 == nnodes(sh))
-		/* Ignore events before node assignment */
-		return ESPANK_SUCCESS;
+	jbuf_t jb = NULL;
 
 	switch (context) {
+	case S_CTX_ALLOCATOR:
+		/* Called from sbatch/salloc which is likely running on the head node. */
```

Comment:
I do not think that we should do anything at all in S_CTX_ALLOCATOR. I think the whole case statement can be removed and just be replace with a simple conditional on S_CTX_REMOTE.

Reply:
agreed, and my testing confirms that this event is not useful for our purposes